Mathlib/NumberTheory/SmoothNumbers.lean

```diff
@@ -49,87 +55,248 @@ lemma primesBelow_succ (n : ℕ) :
 lemma not_mem_primesBelow (n : ℕ) : n ∉ primesBelow n :=
   fun hn ↦ (lt_of_mem_primesBelow hn).false
 
+
+/-!
+### `s`-factored numbers
+-/
+
+/-- `factoredNumbers s`, for a finite set `s` of natural numbers, is the set of positive natural
+numbers all of whose prime factors are in `s`. -/
+def factoredNumbers (s : Finset ℕ) : Set ℕ := {m | m ≠ 0 ∧ ∀ p ∈ factors m, p ∈ s}
+
+lemma mem_factoredNumbers {s : Finset ℕ} {m : ℕ} :
+    m ∈ factoredNumbers s ↔ m ≠ 0 ∧ ∀ p ∈ factors m, p ∈ s :=
+  Iff.rfl
+
+/-- Membership in `Nat.factoredNumbers n` is decidable. -/
+instance (s : Finset ℕ) : DecidablePred (· ∈ factoredNumbers s) :=
+  inferInstanceAs <| DecidablePred fun x ↦ x ∈ {m | m ≠ 0 ∧ ∀ p ∈ factors m, p ∈ s}
+
+/-- A number that divides an `s`-factored number is itself `s`-factored. -/
+lemma mem_factoredNumbers_of_dvd {s : Finset ℕ} {m k : ℕ} (h : m ∈ factoredNumbers s)
+    (h' : k ∣ m) :
+    k ∈ factoredNumbers s := by
```

Comment:
```suggestion
    (h' : k ∣ m) : k ∈ factoredNumbers s := by
```
Isn't that the preferred style?

Reply:
My preference is to have the statement on a separate line when name+args span more than one line.
I've been told in another PR that both versions are fine, so I'll leave it as it is.

Reply:
I'll merge after breakfast unless told not to.